File: telerik_ui_for_winforms_courseware_chm/Courseware/Treeview/VB/DatabindingTreeView/Databinding/SelfReferencing.vb
Public Class SelfReferencing

    Private Sub SelfReferencing_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.employeesTableAdapter.Fill(Me.nwindDataSet.Employees)
        radTreeView1.DataSource = Me.nwindDataSet.Employees
        radTreeView1.DisplayMember = "Title"
        radTreeView1.ChildMember = "EmployeeID"
        radTreeView1.ParentMember = "ReportsTo"
    End Sub
End Class

File: telerik_ui_for_winforms_courseware_chm/Courseware/Treeview/CS/DatabindingTreeView/Databinding/SelfReferencing.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace Databinding
{
    public partial class SelfReferencing : RadForm
    {
        public SelfReferencing()
        {
            InitializeComponent();
        }

        private void RadForm1_Load(object sender, EventArgs e)
        {
            this.employeesTableAdapter.Fill(this.nwindDataSet.Employees);
            radTreeView1.DataSource = this.nwindDataSet.Employees;
            radTreeView1.DisplayMember = "Title";
            radTreeView1.ChildMember = "EmployeeID";
            radTreeView1.ParentMember = "ReportsTo";
        }
    }
}
